### PR TITLE
Introduction to mathematical intuition behind ResNets.

### PR DESCRIPTION
**Pull request by 0xD4rky**
Hi Maintainers,

I have updated the ResNet documentation in Unit-2 (CNNs) and I have tried to include the following points in my documentation:

1. How  "residual learning" allows networks to learn easier-to-optimize residual functions rather than directly mapping inputs to outputs.
2. How ResNets solve the degradation problem
3. How ResNets improve gradient flow by ensuring that gradients can propagate back through residual connections
4. How its possible to train "deeper neural networks" using ResNets.

Please have a look at my updated documentation and inform me if there is a need to better the documentation changes that I have done. Looking forward to it!

### CODE REVIEW

File: chapters/en/unit2/cnns/resnet.mdx [hunk 1]
```diff
@@ -22,16 +24,52 @@ ResNet’s residual connections unlocked the potential of the extreme depth, pro
...
+
+
+Let's break this down:
+
+#### 1. Basic Building Block: Residual Block
+
+In a typical neural network layer, we aim to learn a mapping F(x), where x is the input and F is the transformation the network applies. Without residual learning, the transformation at a layer is: y = F(x). In ResNets, instead of learning F(x) directly, the network is designed to learn the residual R(x), where: R(x) = F(x) − x. Thus, the transformation in a residual block is written as: y = F(x) + x.
+
+Here, x is the input to the residual block, and F(x) is the output of the block's stacked layers (usually convolutions, batch normalization, and ReLU). The identity mappingx is directly added to the output of F(x) (through the skip connection). So the block is learning the residual R(x) = F(x), and the final output is F(x) + x. This residual function is easier to optimize compared to the original mapping. If the optimal transformation is close to the identity (i.e., no transformation is needed), the network can easily set F(x) ≈ 0 to pass through the input as it is.
```
Comment (ATaylorAerospace) [inline chapters/en/unit2/cnns/resnet.mdx:36]:
```suggestion
Here, x is the input to the residual block, and F(x) is the output of the block's stacked layers (usually convolutions, batch normalization, and ReLU). The identity mapping x is directly added to the output of F(x) (through the skip connection). So the block is learning the residual R(x) = F(x), and the final output is F(x) + x. This residual function is easier to optimize compared to the original mapping. If the optimal transformation is close to the identity (i.e., no transformation is needed), the network can easily set F(x) ≈ 0 to pass through the input as it is.
```

**Review by ATaylorAerospace — COMMENTED** [review]
One minor nit. Thanks for adding the additional content.

**Comment by 0xD4rky** [comment]
> One minor nit. Thanks for adding the additional content.

thanks for spotting it!

**Review by ATaylorAerospace — APPROVED** [review]

**Review by sergiopaniego — APPROVED** [review]

File: chapters/en/unit2/cnns/resnet.mdx [hunk 2]
```diff
@@ -12,6 +12,8 @@ An obstacle to answering this question, the gradient vanishing problem, was addr
 
 However, a new issue emerged: the degradation problem. As the neural networks became deeper, accuracy saturated and degraded rapidly. An experiment comparing shallow and deep plain networks revealed that deeper models exhibited higher training and test errors, suggesting a fundamental challenge in training deeper architectures effectively. This degradation was not because of overfitting but because the training error increased when the network became deeper. The added layers did not approximate the identity function.
 
+<img src="https://i.sstatic.net/xDLez.jpg" alt="Degradation-error" width="512" height="512">
```
Comment (johko) [inline chapters/en/unit2/cnns/resnet.mdx:15]: in general we try to use this dataset/repo for all assets: https://huggingface.co/datasets/hf-vision/course-assets
You can create a PR in the dataset with the image you want to add and then reference it here like:
`residual` (but of course with your image name)

File: chapters/en/unit2/cnns/resnet.mdx [hunk 3]
```diff
@@ -22,16 +24,52 @@ ResNet’s residual connections unlocked the potential of the extreme depth, pro
...
+`∂x/∂L = ∂F(x)/∂L * ∂F(x)/∂x`
+
+However, for the residual block, where y = F(x) + x, the gradient becomes:
+
+`∂L/∂x = ∂L/∂F(x) * ∂F(x)/∂x + ∂L/∂F(x) ⋅1`
+
+Notice that we now have an additional term 1 in the gradient calculation. This means that the gradient at each layer has a direct path back to earlier layers, improving gradient flow and reducing the chance of vanishing gradients. The gradients flow more easily through the network, allowing deeper networks to train without degradation.
+
+
+
```
Comment (johko) [inline chapters/en/unit2/cnns/resnet.mdx:58]: I don't think we need that many blank lines ;) 
```suggestion

```

**Review by johko — COMMENTED** [review]
This made me think about residuals a bit more, thanks for the added content.
Just two smallish things to edit (the image thing is mainly for consistency within the course)

**Comment by 0xD4rky** [comment]
Thanks @johko, will keep the above considerations in mind from the next PRs.

**Review by 0xD4rky — COMMENTED** [review]
Thanks for changing those issues!

**Review by 0xD4rky — COMMENTED** [review]
looks good to me, will implement these changes

**Review by johko — APPROVED** [review]
Let's just ignore the image URL for now to get this merged. But if you ever have some spare time @0xD4rky , feel free to change the image link

**Comment by 0xD4rky** [comment]
> Let's just ignore the image URL for now to get this merged. But if you ever have some spare time @0xD4rky , feel free to change the image link

my bad @johko, will change them in the future (was busy in exams :/)
